frontend/src/components/App.js

```diff
@@ -6,27 +6,29 @@ import './App.css';
 import { initialState, reducer, AppContext } from './AppState';
 
 import CompareAssignmentsTab from './CompareAssignmentsTab/CompareAssignmentsTab';
+import ErrorCard from './ErrorCard/ErrorCard';
 
 const App = () => {
   const [state, dispatch] = useReducer(reducer, initialState);
 
   useEffect(() => {
     fetch(`data/${window.django.course_id}`)
       .then((response) => {
-        if (response.status >= 400) {
-          return dispatch({ type: businessDataFetchErrored, value: 'The canvas api fetch failed.' });
+        if (response.status >= 400 || response.status === 204) {
```

Comment:
I feel like if there's truly a problem processing the data on the server-side, we should probably be returning an HTTP 500, but maybe 204 is appropriate if the processing is successful but there's actually no data? Perhaps something to revisit.

Reply:
I think I may have delved to deep into semantics here. An application error probably just fits and requires less explanation (so therefore better in the long run).